### PR TITLE
1426. Counting Elements-Two alternative solutions

### PR DESCRIPTION
**Pull request by ErdemT09**
We can actually use the constraints given in the question to store the existence of numbers in a boolean array.  The array is looped once again and we check if number+1 is true in the array. I personally do not really like this solution as it wouldn't be valid if the constraints were something unknown to us and not limited by LeetCode or the programming language itself (For example, `Integer `type in Haskell represents an unbounded integer value).
The other solution is a bit more general, and can be pretty fast as well depending on the implementation of the HashSet. We add all the numbers in the array to a set and after that, check if the sought value is in the set.

### CODE REVIEW

**Comment by altayhunoglu** [comment]
issue: https://github.com/spiralgo/algorithms/issues/150

File: src/main/java/algorithms/curated170/easy/CountingElementsLinear.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            check[n] = true;
+        }
+        for (int n : arr) {
+            if (check[n + 1]) {
+                count++;
+            }
+        }
+        return count;
+    }
+
```
Comment (altayhunoglu) [inline src/main/java/algorithms/curated170/easy/CountingElementsLinear.java:17]: It is quite fine.
image

File: src/main/java/algorithms/curated170/easy/CountingElementsSet.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        for (int n : arr) {
+            nums.add(n);
+        }
+        for (int n : arr) {
+            if (nums.contains(n + 1)) {
+                count++;
+            }
+        }
+        return count;
+    }
```
Comment (altayhunoglu) [inline src/main/java/algorithms/curated170/easy/CountingElementsSet.java:20]: This is quite fine as well.
image

**Review by altayhunoglu — APPROVED** [review]
Thank you for these nice solutions Erdem.

File: src/main/java/algorithms/curated170/easy/CountingElementsSet.java (hunk 2 above)
Comment (ErdemT09) [reply src/main/java/algorithms/curated170/easy/CountingElementsSet.java:20]: How does the current solution with sorting perform?

Comment (altayhunoglu) [reply src/main/java/algorithms/curated170/easy/CountingElementsSet.java:20]: image

**Pull request merged**